.github/workflows/python-app.yml

```diff
@@ -27,11 +27,11 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-      - name: Lint with pylint
+          python -m pip install uv
+          if [ -f requirements.txt ]; then uv pip install -r requirements.txt --system; fi
+      - name: Lint with ruff and pylint
         run: |
-          pylint packet/routes packet
+          ruff check packet && pylint packet/routes packet
```

Comment:
nit: typically each linter/formatter is run as a separate stage since its easier to tell which one failed